File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque

n, m, v = map(int, input().split())

graph = [[] for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for i in range(1, n + 1):
    graph[i].sort()  

def bfs(start):
    visited = [False] * (n + 1)
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end = ' ')
        for i in graph[v]:
            if not visited[i]:
                visited[i] = True
                queue.append(i)
    
    
dfs_visited = [False] * (n + 1)
def dfs(v):
    dfs_visited[v] = True
    print(v, end=" ")
    for i in graph[v]:
        if not dfs_visited[i]:
            dfs(i)
            
dfs(v)    
print()
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
	static int n, m, v;
	static ArrayList<Integer>[] map;
	static boolean[] dfs_visited;
	static boolean[] bfs_visited;

	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		n = sc.nextInt();
		m = sc.nextInt();
		v = sc.nextInt();
		dfs_visited = new boolean[n + 1];
		bfs_visited = new boolean[n + 1];

		map = new ArrayList[n + 1];
		for (int i = 1; i <= n; i++) {
			map[i] = new ArrayList<>();
		}

		for (int i = 0; i < m; i++) {
			int a = sc.nextInt();
			int b = sc.nextInt();

			map[a].add(b);
			map[b].add(a);
		}

		for (int i = 1; i <= n; i++) {
			map[i].sort(null);
		}
		dfs(v);
		System.out.println();
		bfs(v);
	}

	static void dfs(int index) {
		if (dfs_visited[index] == true) {
			return;
		}
		dfs_visited[index] = true;
		System.out.print(index + " ");
		for (int next : map[index]) {
			dfs(next);
		}
	}

	static void bfs(int node) {
		Queue<Integer> q = new LinkedList<>();
		q.offer(node);
		bfs_visited[node] = true;
		System.out.print(node + " ");
		while (!q.isEmpty()) {
			int now = q.poll();
			for (int next : map[now]) {
				if (bfs_visited[next]) {
					continue;
				}
				System.out.print(next + " ");
				q.offer(next);
				bfs_visited[next] = true;
			}
		}

	}
}